File: Projects/Project_07_3D_Elliptic/For_Instructors/julia/warp_grid.jl
#---------------------------------------------------------------------#
#This function warps a 2D grid
#Written by F.X. Giraldo on 5/2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function warp_grid!(coord,Npoin,DFloat)

    for i=1:Npoin
        x, y, z = coord[1,i], coord[2,i], coord[3,i]
        coord[1,i]=x + sin(π * x) * sin(2 * π * y) * sin(2 * π * z) / 50
        coord[2,i]=y + sin(2 * π * x) * sin(π * y) * sin(2 * π * z) / 50
        coord[3,i]=z + sin(2 * π * x) * sin(2 * π * y) * sin(π * z) / 50
    end

end

File: Projects/Project_05_2D_Elliptic/For_Instructors/matlab/warp_grid.m
%---------------------------------------------------------------------%
%This function warp the grid and plots it.
%Written by F.X. Giraldo on 4/2008
%           Department of Applied Mathematics
%           Naval Postgraduate School 
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function coord = warp_grid(coord,npoin)

c=1/15;

%Warp Grid
for i=1:npoin
    x=coord(1,i);
    y=coord(2,i);
    coord(1,i)=x + sin(pi * x) * sin(2 * pi * y)*c;
    coord(2,i)=y + sin(2 * pi * x) * sin(pi * y)*c;
    % coord(1,i)=x + sin(pi * x) * cos(2 * pi * y)*c;
    % coord(2,i)=y + sin(pi * y) * cos(2 * pi * x)*c;
end




      


      
